sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_options.hpp

```diff
@@ -209,6 +209,115 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     Models::FileSmbProperties SmbProperties;
   };
 
+  /**
+   * @brief Optional parameters for
+   * #Azure::Storage::Files::Shares::ShareDirectoryClient::RenameFile.
+   */
+  struct RenameFileOptions final
+  {
+    /**
+     * A boolean value for if the destination file already exists, whether this request will
+     * overwrite the file or not. If true, the rename will succeed and will overwrite the
+     * destination file. If not provided or if false and the destination file does exist, the
+     * request will not overwrite the destination file. If provided and the destination file doesn�t
+     * exist, the rename will succeed.
+     */
+    Azure::Nullable<bool> ReplaceIfExists;
+
+    /**
+     * A boolean value that specifies whether the ReadOnly attribute on a preexisting destination
+     * file should be respected. If true, the rename will succeed, otherwise, a previous file at the
+     * destination with the ReadOnly attribute set will cause the rename to fail. ReplaceIfExists
+     * must also be true.
+     */
+    Azure::Nullable<bool> IgnoreReadOnly;
+
+    /**
+     * Specify the access condition for the path.
+     */
+    LeaseAccessConditions AccessConditions;
+
+    /**
+     * The access condition for source path.
+     */
+    LeaseAccessConditions SourceAccessConditions;
+
+    /**
+     * SMB properties to set for the directory.
+     */
+    Models::FileSmbProperties SmbProperties;
+
+    /**
+     * If specified the permission (security descriptor) shall be set for the directory.
+     * This option can be used if Permission size is <= 8KB, else SmbProperties.PermissionKey
+     * shall be used.A value of preserve may be passed to keep an existing value unchanged.
+     */
+    Azure::Nullable<std::string> FilePermission;
+
+    /**
+     * A name-value pair to associate with a file storage object.
+     */
+    Storage::Metadata Metadata;
+
+    /**
+     * Sets the file's content type.If this property is not specified on the request, then the
+     * property will be preserved for the file
+     */
+    Nullable<std::string> FileContentType;
```

Comment:
We have `FileHttpHeaders` and this field is a member of the struct. I don't understand why we support only this one field. It would be awkward if in the future we add support for the other fields.

I think we should remove this field for now.

Reply:
yeah, remove this header.